paperoni/commands/command_collect_semantic_scholar.py

```diff
@@ -0,0 +1,281 @@
+from coleo import Option, default, tooled
+
+from ..papers2 import Paper
+from .interactive import InteractiveCommands, default_commands
+from .command_semantic_scholar import search
+from ..sql.database import Database
+from ..utils import get_venue_name_and_volume
+
+
+class Collection:
+    __slots__ = ("db", "papers_added", "papers_excluded")
+
+    def __init__(self, filename):
+        self.db = Database(filename)
+        self.papers_added = set()
+        self.papers_excluded = set()
+
+    def __contains__(self, paper: Paper):
+        if paper in self.papers_excluded:
+            return False
+        elif paper in self.papers_added:
+            return True
+        else:
+            return self.db.select_id_from_values(
+                "paper", "paper_id", title=paper.title, abstract=paper.abstract
+            )
+
+    def add(self, paper: Paper):
+        if paper in self.papers_excluded:
+            self.papers_excluded.remove(paper)
+        self.papers_added.add(paper)
+
+    def exclude(self, paper: Paper):
+        if paper in self.papers_added:
+            self.papers_added.remove(paper)
+        self.papers_excluded.add(paper)
+
+    def excludes(self, paper: Paper):
+        return paper in self.papers_excluded
+
+    def save(self):
+        # Delete excluded papers.
+        for paper in self.papers_excluded:
+            if paper.abstract is None:
+                self.db.modify(
+                    "DELETE FROM paper WHERE title = ? AND abstract IS NULL",
+                    [paper.title],
+                )
+            else:
+                self.db.modify(
+                    "DELETE FROM paper WHERE title = ? AND abstract = ?",
+                    (paper.title, paper.abstract),
+                )
```

Comment:
1. Why not use the paper's ID?
2. Excluding a paper should normally prevent adding it at a later date, if e.g. we perform bulk additions based on a search for an author's papers, we should not re-add publications we had excluded for a reason or another. This means we should keep the entry and not delete it, but mark it as "excluded". This could be an extra column in the paper table, we will just have to be careful to add the appropriate where clause when listing papers.